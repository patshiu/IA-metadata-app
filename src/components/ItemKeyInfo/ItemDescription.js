import React from 'react';
import PropTypes from 'prop-types';

const ItemDescription = ({ description }) => {
  // Content fetched from known MAPI, should be safe.
  return (
    <div data-testid="item-description">
      <div dangerouslySetInnerHTML={{ __html: description }} />
    </div>
  );
};

ItemDescription.propTypes = {
  description: PropTypes.oneOfType([PropTypes.string, PropTypes.array])
};

export default ItemDescription;
