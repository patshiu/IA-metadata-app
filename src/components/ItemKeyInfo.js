import React from 'react';
import PropTypes from 'prop-types';

import ItemTitle from './ItemTitle';
import ItemDescription from './ItemDescription';

const ItemKeyInfo = ({ title, description }) => {
  return (
    <div className="row">
      <div className="section">
        <ItemTitle title={title} />
        <ItemDescription description={description} />
      </div>
    </div>
  );
};

ItemKeyInfo.propTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string
};

export default ItemKeyInfo;
