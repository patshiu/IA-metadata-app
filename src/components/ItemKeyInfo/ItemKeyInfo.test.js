import React from 'react';
import renderer from 'react-test-renderer';

import ItemKeyInfo from './ItemKeyInfo';

it('renders item key info correctly', () => {
  const description =
    '<h2>This a description of an item.</h2> <p>It may contain <a href="https://developer.mozilla.org/en-US/docs/Learn/Getting_started_with_the_web/HTML_basics#">html</a>.</p>';
  const title = 'This Is An Item Title';
  const tree = renderer.create(<ItemKeyInfo title={title} description={description} />).toJSON();
  expect(tree).toMatchSnapshot();
});
