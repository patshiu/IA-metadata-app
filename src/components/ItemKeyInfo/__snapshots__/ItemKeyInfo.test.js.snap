// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders item key info correctly 1`] = `
<div
  className="row"
  data-testid="item-key-info"
>
  <div
    className="section"
  >
    <h1>
      This Is An Item Title
    </h1>
    <div
      data-testid="item-description"
    >
      <div
        dangerouslySetInnerHTML={
          Object {
            "__html": "<h2>This a description of an item.</h2> <p>It may contain <a href=\\"https://developer.mozilla.org/en-US/docs/Learn/Getting_started_with_the_web/HTML_basics#\\">html</a>.</p>",
          }
        }
      />
    </div>
  </div>
</div>
`;
